internal/healthcheck/healthcheck.go

```diff
@@ -91,28 +97,29 @@ func (hs *HealthServer) Shutdown(ctx context.Context) error {
 }
 
 // UpstreamStats calculates health stats for the host identified by the hostname in the request URL.
-func (hs *HealthServer) UpstreamStats(w http.ResponseWriter, r *http.Request) {
+func (hs *HealthServer) UpstreamStats(w http.ResponseWriter, r *http.Request, cnf *configs.Configurator) {
```

Comment:
how about using the configurator on the server struct instead of changing handler's signature?